contracts/stellar-interchain-token-service/src/tests/migrate.rs

```diff
@@ -1,19 +1,199 @@
+            }
+            TokenManagerType::LockUnlock => {
+                let token = env.register_stellar_asset_contract_v2(owner.clone());
+                token_id = its_client.register_canonical_token(&token.address());
+            }
+        }
+
+        let its_wasm_hash = env
+            .deployer()
+            .upload_contract_wasm(NEW_INTERCHAIN_TOKEN_SERVICE_WASM);
+        let new_token_manager_wasm_hash =
+            env.deployer().upload_contract_wasm(TOKEN_MANAGER_WASM_V110);
+        let new_interchain_token_wasm_hash = env
+            .deployer()
+            .upload_contract_wasm(INTERCHAIN_TOKEN_WASM_V110);
+
+        let current_epoch = current_epoch(&env);
+
+        MigrateTestConfig {
+            env,
+            owner,
+            its_client,
+            upgrader_client,
+            token_id,
+            current_epoch,
+            its_wasm_hash,
+            migration_data: CustomMigrationData {
+                new_token_manager_wasm_hash,
+                new_interchain_token_wasm_hash,
+            },
+        }
+    }
+
+    pub fn setup_migrate_storage(
+        env: &Env,
+        its_client: &InterchainTokenServiceClient<'_>,
+        token_id: BytesN<32>,
+        current_epoch: u64,
+        flow_in_amount: i128,
+        flow_out_amount: i128,
+    ) {
+        env.as_contract(&its_client.address, || {
+            let flow_key = legacy_storage::FlowKey {
+                token_id: token_id.clone(),
+                epoch: current_epoch,
+            };
+
+            legacy_storage::set_flow_in(env, flow_key.clone(), &flow_in_amount);
+            legacy_storage::set_flow_out(env, flow_key, &flow_out_amount);
+        });
+    }
+
+    pub fn upgrade<'a>(
+        env: &Env,
+        owner: Address,
+        its_client: &InterchainTokenServiceClient<'a>,
+        upgrader_client: &UpgraderClient<'a>,
+        its_wasm_hash: BytesN<32>,
+        migration_data: CustomMigrationData,
+    ) -> std::string::String {
+        let its_upgrade_auth = mock_auth!(owner, its_client.upgrade(&its_wasm_hash));
+        let its_migrate_auth = mock_auth!(owner, its_client.migrate(migration_data));
+        let upgrader_upgrade_auth = mock_auth!(
+            owner,
+            upgrader_client.upgrade(
+                &its_client.address,
+                &String::from_str(env, NEW_VERSION),
+                &its_wasm_hash,
+                &vec![&env, migration_data.clone()],
+            ),
+            &[
+                its_upgrade_auth.invoke.clone(),
+                its_migrate_auth.invoke.clone()
+            ]
+        );
+
+        upgrader_client
+            .mock_auths(&[upgrader_upgrade_auth])
+            .upgrade(
+                &its_client.address,
+                &String::from_str(env, NEW_VERSION),
+                &its_wasm_hash,
+                &vec![&env, migration_data.into_val(env)],
+            );
+
+        format_auths(env.auths(), "upgrader.upgrade(...)")
```

Comment:
this is not needed since we're using `mock_auths` already. Also, we should only save auth for the method that's being tested in the test (rather than save auths for helper calls)

Reply:
the caller can call this where needed